src/test/scala/sangria/macros/derive/DeriveEnumTypeMacroSpec.scala

```diff
@@ -31,6 +39,11 @@ class DeriveEnumTypeMacroSpec extends WordSpec with Matchers {
   case object MegaOrangeAnnotated extends FruitAnnotated
 
   "Enum derivation" should {
+    "support nested enum instances"  in {
+      val enum = deriveEnumType[Difficulty]()
+      enum.name should be ("Difficulty")
```

Comment:
It would be great if you could add assertions for all 3 enum values as well (that there are only 3 available their names are correct). I often had this issue where `knownDirectSubclasses` does not find all of the subtypes, so I normally always try to verify values as well.